### PR TITLE
Add S3 and Azure Blob instructions

### PR DESCRIPTION
**Pull request by ianstanton**
- Add S3 registry backend instructions
- Add Azure Blob Storage registry backend instructions
- Related to  astronomer/issues#1389

### CODE REVIEW

**Comment by paolaperaza** [comment]
@ianstanton This looks good, I just dded an `## Overview` section and an extra sentence that gives more context to how docker images are created.

Two notes:

- Stupid question, but when you make changes to your `config.yaml` - how to you "save"/"inject" them? Is that something we should specify?
- Is there more so say about when a user might enable any of these? Is it really just a storage concern? Does the K8s PVC solution have a quantifiable storage limit? Mostly wondering if we could give more specific guidelines on when we'd recommend configuring any of these solutions.

Thanks for doing this!

**Comment by ianstanton** [comment]
- You would apply these with `helm upgrade` if your platform is already up and running, otherwise they will be applied on initial run of `helm install`. I can add a note if you like.
- This is mostly a storage concern, but also a best practice I feel we should recommend. I see this becoming more common going forward as we've already had a few customers fill up the default storage allocation (we'll probably add S3 registry back end as a default in our AWS terraform).  If it is part of our validated designs we will want to mention something about that here.

**Comment by ianstanton** [comment]
@paolaperaza Are there any changes that need to be made here? If you'd like me to add what you mentioned above just let me know.

**Comment by paolaperaza** [comment]
Ah, sorry @ianstanton ! I'd say we can add a sentence clarifying that this is our recommendation (if it is) and not just something "extra" for people who really want it. Want to make that change?

I feel less strongly about the direction on `helm upgrade` - seems like we don't do that in other docs, so I'm ok not including it if you think it's superfluous- will defer 2 u.

**Comment by vparekh94** [comment]
@paolaperaza @ianstanton we should definitely add it here for now. I think we'll need a doc that goes over applying a `helm upgrade` that we can link to from other docs, but until that doc exists, we can't really expect a customer to use this without those directions.

**Comment by paolaperaza** [comment]
@ianstanton I just pushed up a commit that hopefully makes it a bit more clear up top that we'd recommend this solution and added an "Apply your Changes" section following instructions for users to update their `config.yaml`.

This accurate? ⤵️ 

> From here, find your platform release name and apply your changes.
> 
> ```
> $ helm ls --namespace <your-platform-namespace>
> ```
> 
> ```
> $ helm upgrade <platform-release-name> -f config.yaml . --namespace <your-namespace>

Let me know if this works (or not)!

**Comment by ianstanton** [comment]
@paolaperaza as of 0.12 it should look like this:
`helm upgrade <platform-release-name> -f config.yaml --version=<platform-version> astronomer/astronomer -n <your-namespace>`
Example:
`helm upgrade astronomer -f config.yaml --version=0.15.5 astronomer/astronomer -n astronomer`

**Comment by paolaperaza** [comment]
@ianstanton Ah, got it! Whoops, got that from another doc so turns out that command is wrong in a bunch of places. Corrected it in all the 0.15 docs I could find.

Want to take a final look?

File: v0.15/ee-registry-backend.md [hunk 1]
```diff
@@ -60,6 +84,119 @@ astronomer:
...
+```
+
+```
+$ helm upgrade <platform-release-name> -f config.yaml --version=<platform-version> astronomer/astronomer -n <your-namespace>
+```
+
+For example, if you're running Astronomer v0.15.5 and your platform release name is `astronomer`, you might run:
+
+```
+$ helm upgrade astronomer -f config.yaml --version=<platform-version> astronomer/astronomer -n astronomer
```
Comment (ianstanton) [inline v0.15/ee-registry-backend.md:105]: `0.15.5` instead of `<platform-version>` in this example

File: v0.15/ee-registry-backend.md [hunk 2]
```diff
@@ -60,6 +84,119 @@ astronomer:
...
```
Comment (ianstanton) [inline v0.15/ee-registry-backend.md:153]: Same as above (version)

File: v0.15/ee-registry-backend.md [hunk 3]
```diff
@@ -60,6 +84,119 @@ astronomer:
...
+```
+
+```
+$ helm upgrade <platform-release-name> -f config.yaml --version=<platform-version> astronomer/astronomer -n <your-namespace>
+```
+
+For example, if you're running Astronomer v0.15.5 and your platform release name is `astronomer`, you might run:
 
 ```
+$ helm upgrade astronomer -f config.yaml --version=<platform-version> astronomer/astronomer -n astronomer
```
Comment (ianstanton) [inline v0.15/ee-registry-backend.md:201]: Same as above (version)

**Comment by paolaperaza** [comment]
Ah! Thanks @ianstanton - fixed. Merging 😊

**Pull request merged**